### PR TITLE
Add instructions for how to set up toolchain in an IDE to docs (currently only CLion)

### PR DESCRIPTION
**Pull request by HexCodeFFF**
CLion is able to properly integrate the library into its code assistance toolchain, although it's weird to set up. I thought I'd contribute the results of me poking around my IDE and leave space for people to add instructions for other IDEs. I genuinely can't figure out VS code so currently, it's only for CLion

### CODE REVIEW

**Comment by mateoconlechuga** [comment]
Thanks for the PR! I am a little concerned about this being windows-only - do you know if it works differently on linux/mac? Can you add a note that you do not have to set the `make.exe` part on linux/mac since it uses the system one?

**Comment by HexCodeFFF** [comment]
Oh I just assumed since everything was an exe that it was windows only oops

currently I don’t have any non-windows machines to test on so I have no idea

is the included make.exe not a custom thing?

**Comment by HexCodeFFF** [comment]
ok so yeah you're right the macos/linux ones don't come with a make executable and it gets the information from the system path. I'll update that.

**Comment by mateoconlechuga** [comment]
The included make is just the standard make executable for windows - but make is a general system development application on other OSes. Each toolchain releases differs slightly based on the OS. I believe as long as you add a comment that the make executable uses the system one on Linux/Mac, and that there is no `.exe` extension on those OSes, everything else looks good.

**Comment by mateoconlechuga** [comment]
Also since CLion is generally paid, I'm not sure how other people might feel about this, so just keep that in mind. I've heard sublime text is also pretty nice.

**Comment by HexCodeFFF** [comment]
I got CLion free through the GitHub Student Pack. I tried to poke around in Visual Studio to set it up but I am not familiar with that IDE at all so I couldn't figure it out. Also, Sublime is just a text editor so I don't think it has any built-in code assistance that can be set up

**Comment by mateoconlechuga** [comment]
"sublime is just a text editor" lolololol

Anyway, just make those above changes and then I can merge.

**Comment by HexCodeFFF** [comment]
> "sublime is just a text editor" lolololol - you can do pretty much anything with it.

I've never personally used it, and quick googling suggested it wasn't capable of code assistance, but I'm probably wrong about that :p

Either way, I leave adding more IDEs to someone actually familiar with the IDE in question.

File: docs/static/ides.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+.. image:: images/ides_clion_8.png
+   :align: center
+
+Then add :code:`debug` to the arguments, and rename it to distinguish it from the normal one.
+
+.. image:: images/ides_clion_9.png
+   :align: center
+
+You can now switch between the normal and debug configurations in the rop right.
```
Comment (mateoconlechuga) [inline docs/static/ides.rst:77]: What is the "rop right" :laughing:

File: docs/static/ides.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+Set the "Build Tool" to :code:`CEdev/bin/make.exe` and both C compilers to :code:`CEdev/bin/ez80-clang.exe`
+
+MacOS/Linux come with their own system Make executable, so you don't need to edit that.
+
+On MacOS/Linux, executables have no file extension. Set both C compilers to :code:`CEdev/bin/ez80-clang`.
+
+.. image:: images/ides_clion_4.png
+   :align: center
+
+Since this compiler is highly unusual, CLion will warn that "Test CMake run finished with errors". This is safe to ignore.
```
Comment (mateoconlechuga) [inline docs/static/ides.rst:38]: "highly unusual" doesn't tell me anything here

**Review by mateoconlechuga — CHANGES_REQUESTED** [review]

**Review by mateoconlechuga — APPROVED** [review]

**Pull request merged**